src/adapters/networkhandler/mockNetworkHandler.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+const path = require('path');
+const { getIntegrations } = require("../../routes/utils");
+// const { SUPPORTED_VERSIONS } = require("../../routes/utils/constants");
+
+const mockResponses = {};
+
+// Dynamically import the network handlers for all
+// the supported destinations
+const destinations = getIntegrations(path.resolve(__dirname, `../mocks`));
+console.log(`Destinations for proxy: ${destinations}`);
+destinations.forEach((dest) => {
+  mockResponses[dest] = require(`../mocks/${dest}/response.json`);
+});
+
+
+const sleep = async (sleepInMs) => new Promise((resolve) => { setTimeout(resolve, sleepInMs) });
+
+function mockNetworkHandler (type, originalNetworkHandler) {
+  const destMockResponses = mockResponses[type];
+
+  return function networkHandler() {
+    this.proxy = async (request) => {
+      if(!destMockResponses?.length || destMockResponses?.length === 0) {
+        // No destMockResponses available or if the destMockResponses is empty
+        return originalNetworkHandler?.proxy(request);
```

Comment:
Since this is mock handler should we call the original destination even though the mocked responses are not there?

Reply:
This is like a fallback for the scenario wherein we don't have mock data available

Reply:
but then we should just return with something generic in mock environment we should not be sending to dests

Reply:
As we don't have what kind of mock response should be returned we are sending it to destination itself as something generic can cause issues(in terms of interaction with server or in transformer itself)

Reply:
How will we handle duplicate scenarios?
Maybe I am missing something

Reply:
I don't understand what you mean here. Can you elaborate what did you mean ?

Reply:
Ok I checked the diagram again in the notion basically we are either running proxy mock mode or proxy mode not both in any circumstances ok then it should be fine

Reply:
Yes. The intention is to run in a single mode at any point of time